sdk/Sdk.Analyzers/BindingTypeNotSupported.cs

```diff
@@ -0,0 +1,133 @@
+                if (allowConverterFallbackParameterValue is bool allowFallback && allowFallback)
+                {
+                    // If allowConverterFallback is true, we don't need to check for supported types
+                    // because we don't know all of the types that are supported via the fallback
+                    continue;
+                }
+
+                var supportedTypes = GetSupportedTypes(context, inputConverterAttributes);
+                if (supportedTypes.Count <= 0 || supportedTypes.Contains(parameter.Type))
+                {
+                    continue;
+                }
+
+                ReportDiagnostic(context, parameter, attributeType);
+            }
+        }
+
+        private static List<AttributeData> GetInputConverterAttributes(SymbolAnalysisContext context, ITypeSymbol attributeType)
+        {
+            var inputConverterAttributeType = context.Compilation.GetTypeByMetadataName(Constants.Types.InputConverterAttribute);
+            return attributeType.GetAttributes()
+                .Where(attr => attr.AttributeClass.Equals(inputConverterAttributeType))
+                .ToList();
+        }
+
+        private static object GetAllowConverterFallbackParameterValue(SymbolAnalysisContext context, ITypeSymbol attributeType)
+        {
+            var allowConverterFallbackAttributeType = context.Compilation.GetTypeByMetadataName(Constants.Types.AllowConverterFallbackAttribute);
+            var allowConverterFallbackAttribute = attributeType.GetAttributes().FirstOrDefault(attr => attr.AttributeClass.Equals(allowConverterFallbackAttributeType));
+            return allowConverterFallbackAttribute.ConstructorArguments.FirstOrDefault().Value;
+        }
+
+        private static List<object> GetSupportedTypes(SymbolAnalysisContext context, List<AttributeData> inputConverterAttributes)
+        {
+            var supportedTypes = new List<object>();
+
+            foreach (var inputConverterAttribute in inputConverterAttributes)
+            {
+                var converterName = inputConverterAttribute.ConstructorArguments.FirstOrDefault().Value.ToString();
+                var converter = context.Compilation.GetTypeByMetadataName(converterName);
+
+                var converterAttributes = converter.GetAttributes();
+
+                var converterHasSupportedTypeAttribute = converterAttributes.Any(a => a.AttributeClass.Name == Constants.Names.SupportedConverterTypeAttribute);
+                if (!converterHasSupportedTypeAttribute)
+                {
+                    // If a converter does not have the `SupportedConverterTypeAttribute`, we don't need to check for supported types
+                    continue;
+                }
+
+                supportedTypes.AddRange(converterAttributes
+                    .Where(a => a.AttributeClass.Name == Constants.Names.SupportedConverterTypeAttribute)
```

Comment:
I haven't written an analyzer before but is it okay to be comparing on just name. Should namespace be included in the comparison also?

Reply:
I'll switch to checking the named symbol type to be safe